src/autocomplete/databases/postgresql/postgresql-extract-statements.ts

```diff
@@ -0,0 +1,102 @@
+import {Token, tokenize} from '../../shared/tokenize';
+import {postgreSqlAutocompleteData} from './postgresql-autocomplete';
+import {PostgreSqlLexer} from './generated/PostgreSqlLexer';
+import {CursorPosition} from '../../shared';
+
+export interface PostgreSqlStatement {
+    startLine: number;
+    endLine: number;
+    startColumn: number;
+    endColumn: number;
+    statement: string;
+    hasCursor?: boolean;
```

Comment:
Why do we need this field?

Reply:
This field will allow us to detect which statement we can highlight in websql